packages/cpp/src/browser/cpp-build-configurations.ts

```diff
@@ -33,18 +35,28 @@ class SavedActiveBuildConfiguration {
     configName?: string;
 }
 
+const namePlaceholder = '<Enter Name>';
+const directoryPlaceholder = '<Enter Directory>';
+
 /**
  * Entry point to get the list of build configurations and get/set the active
  * build configuration.
  */
 @injectable()
 export class CppBuildConfigurationManager {
-    @inject(StorageService)
-    protected readonly storageService: StorageService;
 
     @inject(CppPreferences)
     protected readonly cppPreferences: CppPreferences;
 
+    @inject(EditorManager)
+    protected readonly editorManager: EditorManager;
+
+    @inject(StorageService)
+    protected readonly storageService: StorageService;
+
+    @inject(WorkspacePreferenceProvider)
+    protected readonly preferenceProvider: WorkspacePreferenceProvider;
```

Comment:
clients should use `PreferenceService`, `PreferenceProvider` is the implementation details

Reply:
@akosyakov in my implementation I require the use of the `workspace preferences uri` ({workspace}/.theia/settings.json) which was previously provided by the `WorkspacePreferenceProvider`. Do you know of any easy way to obtain the correct uri, or should I instead update both the `PreferenceProvider` and `PreferenceService` to provide a source variable as to where the preferences are located?